python/tests/expressions/test_visitors.py

```diff
@@ -1429,3 +1432,162 @@ def test_rewrite_bound():
             accessor=Accessor(position=0, inner=None),
         )
     )
+
+
+@pytest.fixture
+def spec() -> PartitionSpec:
+    return PartitionSpec(PartitionField(1, 1000, IdentityTransform(), "id"))
+
+
+def test_identity_projection(schema: Schema, spec: PartitionSpec):
+    predicates = [
+        NotNull("id"),
+        IsNull("id"),
+        LessThan("id", 100),
+        LessThanOrEqual("id", 101),
+        GreaterThan("id", 102),
+        GreaterThanOrEqual("id", 103),
+        EqualTo("id", 104),
+        NotEqualTo("id", 105),
+    ]
+
+    expected = [
+        BoundNotNull(
+            term=BoundReference[int](
+                field=NestedField(field_id=1000, name="id", field_type=IntegerType(), required=False),
+                accessor=Accessor(position=0, inner=None),
+            )
+        ),
+        BoundIsNull(
+            term=BoundReference[int](
+                field=NestedField(field_id=1000, name="id", field_type=IntegerType(), required=False),
+                accessor=Accessor(position=0, inner=None),
+            )
+        ),
+        BoundLessThan(
+            term=BoundReference[int](
+                field=NestedField(field_id=1000, name="id", field_type=IntegerType(), required=False),
+                accessor=Accessor(position=0, inner=None),
+            ),
+            literal=literal(100),
+        ),
+        BoundLessThanOrEqual(
+            term=BoundReference[int](
+                field=NestedField(field_id=1000, name="id", field_type=IntegerType(), required=False),
+                accessor=Accessor(position=0, inner=None),
+            ),
+            literal=literal(101),
+        ),
+        BoundGreaterThan(
+            term=BoundReference[int](
+                field=NestedField(field_id=1000, name="id", field_type=IntegerType(), required=False),
+                accessor=Accessor(position=0, inner=None),
+            ),
+            literal=literal(102),
+        ),
+        BoundGreaterThanOrEqual(
+            term=BoundReference[int](
+                field=NestedField(field_id=1000, name="id", field_type=IntegerType(), required=False),
+                accessor=Accessor(position=0, inner=None),
+            ),
+            literal=literal(103),
+        ),
+        BoundEqualTo(
+            term=BoundReference[int](
+                field=NestedField(field_id=1000, name="id", field_type=IntegerType(), required=False),
+                accessor=Accessor(position=0, inner=None),
+            ),
+            literal=literal(104),
+        ),
+        BoundNotEqualTo(
+            term=BoundReference[int](
+                field=NestedField(field_id=1000, name="id", field_type=IntegerType(), required=False),
+                accessor=Accessor(position=0, inner=None),
+            ),
+            literal=literal(105),
+        ),
+    ]
+
+    for idx in range(len(predicates)):
```

Comment:
Isn't it more normal to use `enumerate`?

Reply:
I didn't spend many cycles on this. I changed it into:
```
for predicate, expected in zip(predicates, expected_bound):
    expr = inclusive_projection(predicate, schema, spec)
    assert expr == expected
```
I think the `zip` is nicer than the `enumerate`